### PR TITLE
feat: move life event types and activities to vaults

### PR DESCRIPTION
**Pull request by djaiss**
One year ago I started to work on life events but the direction was not ideal. This PR changes that completely.
Also, it will introduce the real benefit of using Monica: documenting your life as best as possible.

Things that have been done in this PR
- all life events (categories, types and events) are now linked to the vault and not the account. That way, users can have different event types/categories per vault, as some event types might not be shared with other vaults.
- life event types and categories are now configured in the vault settings tab.
- the services to manage life events have been written.

In another PR, we'll add the ability to manage life events for real (vue + viewhelpers, considering that the services are done in this PR).

### CODE REVIEW

File: app/Domains/Contact/ManageLifeEvents/Services/CreateLifeEvent.php [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+    public function rules(): array
+    {
+        return [
+            'account_id' => 'required|integer|exists:accounts,id',
+            'vault_id' => 'required|integer|exists:vaults,id',
+            'author_id' => 'required|integer|exists:users,id',
+            'life_event_type_id' => 'required|integer|exists:life_event_types,id',
+            'summary' => 'nullable|string|max:255',
+            'description' => 'nullable|string|max:65535',
+            'happened_at' => 'date|date_format:Y-m-d',
```
Comment (asbiin) [inline app/Domains/Contact/ManageLifeEvents/Services/CreateLifeEvent.php]: Isn't a `required` field?

File: app/Domains/Contact/ManageLifeEvents/Services/CreateLifeEvent.php [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+            $this->account()->currencies()
+                ->findOrFail($this->data['currency_id']);
+        }
+
+        $this->partipantsCollection = collect();
+        foreach ($this->data['participant_ids'] as $participantId) {
+            $this->partipantsCollection->push(
+                $this->vault->contacts()->findOrFail($participantId)
+            );
+        }
```
Comment (asbiin) [inline app/Domains/Contact/ManageLifeEvents/Services/CreateLifeEvent.php]: What do you think of:
```suggestion
        $this->partipantsCollection = collect($this->data['participant_ids'])
            ->map(fn(int $participantId): Contact => $this->vault->contacts()->findOrFail($participantId));
```

Note: it should be wise to check properly that `participant_ids` contains only `int` values, no?

File: app/Domains/Contact/ManageLifeEvents/Services/UpdateLifeEvent.php [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+            $this->account()->currencies()
+                ->findOrFail($this->data['currency_id']);
+        }
+
+        $this->partipantsCollection = collect();
+        foreach ($this->data['participant_ids'] as $participantId) {
+            $this->partipantsCollection->push(
+                $this->vault->contacts()->findOrFail($participantId)
+            );
+        }
```
Comment (asbiin) [inline app/Domains/Contact/ManageLifeEvents/Services/UpdateLifeEvent.php]:
```suggestion
        $this->partipantsCollection = collect($this->data['participant_ids'])
            ->map(fn(int $participantId): Contact => $this->vault->contacts()->findOrFail($participantId));
```

File: app/Domains/Contact/ManageLifeEvents/Services/UpdateLifeEvent.php [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+    {
+        return [
+            'account_id' => 'required|integer|exists:accounts,id',
+            'vault_id' => 'required|integer|exists:vaults,id',
+            'author_id' => 'required|integer|exists:users,id',
+            'life_event_type_id' => 'required|integer|exists:life_event_types,id',
+            'life_event_id' => 'required|integer|exists:life_events,id',
+            'summary' => 'nullable|string|max:255',
+            'description' => 'nullable|string|max:65535',
+            'happened_at' => 'date|date_format:Y-m-d',
```
Comment (asbiin) [inline app/Domains/Contact/ManageLifeEvents/Services/UpdateLifeEvent.php]: Isn't a `required` field?

**Review by asbiin — APPROVED** [review]
Good job!
See my comments, but there is no blocker ...

File: app/Domains/Contact/ManageLifeEvents/Services/CreateLifeEvent.php (hunk 1 above)
Comment (djaiss) [reply app/Domains/Contact/ManageLifeEvents/Services/CreateLifeEvent.php]: you are right

File: app/Domains/Contact/ManageLifeEvents/Services/CreateLifeEvent.php (hunk 2 above)
Comment (djaiss) [reply app/Domains/Contact/ManageLifeEvents/Services/CreateLifeEvent.php]: It works, dirty bastard.

Comment (djaiss) [reply app/Domains/Contact/ManageLifeEvents/Services/CreateLifeEvent.php]: Changed

File: app/Domains/Contact/ManageLifeEvents/Services/UpdateLifeEvent.php (hunk 4 above)
Comment (djaiss) [reply app/Domains/Contact/ManageLifeEvents/Services/UpdateLifeEvent.php]: it is

**Pull request merged**